community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -129,44 +128,47 @@ public StateHandlingStatementOperations(
 
     // <Cursors>
     @Override
-    public NodeCursor nodeCursor( KernelStatement statement, long nodeId )
+    public Cursor<NodeItem> nodeCursor( KernelStatement statement, long nodeId )
     {
-        NodeCursor cursor = statement.getStoreStatement().acquireSingleNodeCursor( nodeId );
+        Cursor<NodeItem> cursor = statement.getStoreStatement().acquireSingleNodeCursor( nodeId );
         if ( statement.hasTxStateWithChanges() )
         {
-            return statement.txState().augmentSingleNodeCursor( cursor );
+            return statement.txState().augmentSingleNodeCursor( cursor, nodeId );
```

Comment:
Yes, that's true i was thinking more about specific id aware (in this case) cursor type that we can pass into `augmentSingleNodeCursor` which will be used as initialisation source